src/ui_utils.c

```diff
@@ -1022,6 +1022,24 @@ void ui_sidebar_show_hide(void)
 }
 
 
+void ui_update_menu_eol(GeanyDocument *doc)
+{
+	gint eol = sci_get_eol_mode(doc->editor->sci);
+	const gchar *widget_name;
+
+	switch (eol)
+	{
+		case SC_EOL_CR: widget_name = "cr"; break;
+		case SC_EOL_LF: widget_name = "lf"; break;
+		default: widget_name = "crlf"; break;
+	}
+	gtk_check_menu_item_set_active(
+		GTK_CHECK_MENU_ITEM(ui_lookup_widget(main_widgets.window, widget_name)), TRUE);
+
+	ui_update_statusbar(doc, -1);
+}
```

Comment:
There was nothing really deep behind this - just pure laziness - I basically just copied what was in UNDO_ENCODING. Looking at it you are right, this is unnecessary.

To make things consistent I added a similar patch for UNDO_ENCODING too.